plugins/jspsych-spr-moving-window.js

```diff
@@ -399,26 +318,82 @@ var sprMovingWindow = (function(jspsych) {
         const BASE_Y = delta_y; // The height on which lines begin.
         const BASE_X = determineLineHeight(trial_pars.font_family, trial_pars.font_size);
 
-        for (let line = 0; line < lines.length; line++) {
-            liney = BASE_Y + line * delta_y;
-            fragments = lines[line].split(RE_CAP_WHITE_SPACE);
-            fragments = fragments.filter( word => {return word != "";});
-            let runningx = BASE_X;
-            for (let fragment = 0; fragment < fragments.length; fragment++) {
-                let current_fragment = fragments[fragment];
-                let pos = new Pos(runningx, liney);
-                let current_word = new TextInfo(
-                    current_fragment,
-                    pos,
+        let context = {
+            pos : new Pos(BASE_X, BASE_Y),
+        };
+
+        function advanceWhiteSpace(context, ws) {
+            if (ws.get_content() == "\n") {
+                context.pos.x = BASE_X;
+                context.pos.y = context.pos.y + delta_y;
+            }
+            else {
+                let info = new TextInfo(
+                    ws,
+                    new Pos(context.pos.x, context.pos.y),
                     ctx,
                     trial_pars.font_family,
-                    trial_pars.font_size
+                    trial_pars.font_size,
                 );
-                if (!current_word.isWhiteSpace())
-                    words.push(current_word);
-                runningx += current_word.width();
+                context.pos.x += info.width();
             }
         }
+
+        function advanceWord(context, word) {
+            let info = new TextInfo(
+                word,
+                new Pos(context.pos.x, context.pos.y),
+                ctx,
+                trial_pars.font_family,
+                trial_pars.font_size
+            );
+            words.push(info)
+            context.pos.x += info.width();
+        }
+
+        let groups = parsed_stimulus.groups;
+        groups.forEach( // A group can be a whitespace
+            group => {
+                let type = group.get_type();
+                if (type == "WhiteSpace") {
+                    advanceWhiteSpace(context, group);
+                }
+                else if (type == "Group") {
+                    let parts = group.sentence_parts.parts;
```

Comment:
it might be easier to follow if you also have an `advanceGroup` function

Reply:
Yeah, this is a bit of a weird case. This is to allow spaces between groups
```
{this} {would} {not} {be} {possible.}
```
The above must otherwise be written as:
```
{this }{would }{not }{be }{possible.}
```
But adding a advance group should be possible.

Reply:
added the advanceGroup